RInAppMessaging/Classes/CampaignDispatcher.swift

```diff
@@ -74,11 +74,6 @@ internal class CampaignDispatcher: CampaignDispatcherType, TaskSchedulable {
             return
         }
 
-        CommonUtility.lock(resourcesIn: campaignRepository)
```

Comment:
This is the actual fix. Why it was causing an issue? - I don't know (yet)

Reply:
OK. So this was locked and the `decrementImpressionsLeftInCampaign` failed?

Reply:
I think `decrementImpressionsLeftInCampaign` was called but then the campaign list got overridden by some other part of code (I still have to investigate that, it's hard to debug)